read_config.c

```diff
@@ -612,25 +612,108 @@ parse_receiver (struct receiver *receiverp,
 }
 
 int
+expand_port_ranges (argc, argv, exp_argc, exp_argv)
+     int argc;
+     const char **argv;
+     int *exp_argc;
+     const char ***exp_argv;
+{
+  *exp_argc=0;
+  *exp_argv=NULL;
+
+  /* expand port definition ranges */
+  for (int j=0; j<argc; j++)
+  {
+      char *port_begin, *port_end=NULL;
+      int just_copy=0;
+      port_begin=strchr (argv[j], PORT_SEPARATOR);
+      if (port_begin==NULL)
+         just_copy=1;
+      else
+      {
+         char *range_start=NULL, *inc_start=NULL;
+         range_start=strchr (port_begin, '-');
+         inc_start=strchr (port_begin, '+');
+         if (!range_start && !inc_start)
+             just_copy=1;
+         else
+         {
+             int first=atoi (port_begin+1);
+             int last;
+             char *suffix;
+             if (range_start && inc_start)
+             {
+               fprintf (stderr, "Invalid port specification: at %s\n", port_begin);
+               exit (1);
+             }
+             if (range_start)
+             {
+                 last=atoi (range_start+1);
+                 suffix=range_start+1;
+             }
+             if (inc_start)
+             {
+                 last=first+atoi (inc_start+1)-1;
+                 suffix=inc_start+1;
+             }
+             while (*suffix && isdigit (*suffix))
+                 suffix++;
+             for (int k=first;k<=last;k++)
+             {
+                 char *newarg=(char *) malloc (strlen (argv[j]+1));
```

Comment:
I had to convince myself that this length is always sufficient. The (decimal representation of) the port number can become larger by incrementing, but the total length should not exceed the original length when you include the range specification. So that looks fine.